tests/environmentRegistry.feature

```diff
@@ -0,0 +1,36 @@
+
+Feature: register and environments$ methods
+
+
+Scenario: Calling register method registers the environment with the provided envKey and env
+  Given Environment Registry with register method
+  When  User calls register method with by providing a valid envKey and env
+  Then  Registration of the environment with this envKey is performed with success
+
+Scenario: Calling register method without providing the envKey and env
+  Given Environment Registry with register method
+  When  User calls register method without providing the envKey and/or env
+  Then  A relevant error is returned
+
+Scenario: Calling register method with an envKey already registered
+  Given Environment Registry with register method
+  When  User calls register method an envKey and env already registered
+//what do we expect in such case?
+  Then  An error is returned? the existing env is replaced with the new one?
+
```

Comment:
Is there a need for -
Scenario:  Calling register method with an invalid envKey
?

Reply:
An envKey is a string and basically, it will match branch name. 

As I understand now, if you try to register with a non existing key, it will create it. If you try to register with an existing one, it will update it

Reply:
So there isnt any way for an envKey to be invalid?
@MikeSquall

Reply:
If envKey is not a `string` then it is not valid

Reply:
Actually, this should be part of two validation scenarios: one for the `envKey` and one for the `env`.

* `envKey` should be a `string`
* `env` should match this pattern :
``` typescript
interface Env {
  /** Env accessPoints is an Array of url */
  accessPoints: {
    /** The url gives access to a specific version of a project */
    url: string;
  }[];
}
```